File: sol/3066/3066.cpp
#include <queue>

struct Solution {
    using l = long;
    int minOperations(vector<int>& nums, int k) {
        priority_queue<l, vector<l>, std::greater<l>> q;
        for (int x : nums){
            q.push(x);
        }
        int res = 0;
        while (q.top() < k){
            l x = q.top();
            q.pop();
            l y = q.top();
            q.pop();
            q.push(x * 2 + y);
            ++res;
        }
        return res;
    }
};

File: sol/3066/3066.py
import heapq as hq

class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        hq.heapify(nums)
        res = 0
        while nums[0] < k:
            x = hq.heappop(nums)
            hq.heapreplace(nums, x * 2 + nums[0])
            res += 1
        return res

